updated examples and added purchasePlan

meet1993shah opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
updated examples and added purchasePlan

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:
```diff
@@ -2800,6 +2834,10 @@
         },
+        "purchasePlan": {
+          "$ref": "#/definitions/PurchasePlan",
+          "description": "Purchase Plan Information of the Virtual Machine persisted on the Disk."
```
Purchase Plan Information of the image used to create the Disk?

meet1993shah replied:
> Purchase Plan Information of the image used to create the Disk?

Optional field, if the Customer wants to put specific Purchase Plan then they can.

chetmaddula replied:
might need a better description, please ask Vikram about it. Thanks

vikramd-ms replied:
Purchase plan information for the the image from which the OS disk was created.

vikramd-ms replied:
add examples

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json:
```diff
@@ -1800,6 +1804,10 @@
         },
+        "purchasePlan": {
+          "$ref": "#/definitions/PurchasePlan",
+          "description": "Purchase Plan Information of the Virtual Machine persisted on the Disk."
```
if this is snapshot say that purchase plan information copied from source disk/snapshot

vikramd-ms replied:
Purchase plan information for the image from which the source disk for the snapshot was originally created.

meet1993shah committed:
Update disk.json

updated description.

vikramd-ms commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json:
```diff
@@ -25,6 +25,12 @@
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
typo. Examples should be generated by running tests

vikramd-ms commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json:
```diff
@@ -41,6 +47,12 @@
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
typo

vikramd-ms commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json:
```diff
@@ -17,6 +17,12 @@
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
typo

meet1993shah committed:
Update CreateAManagedDiskFromAPlatformImage.json

Update example for creating a managed disk from platform image having purchase plan information

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

update Get Disk with disk having purchase plan information

meet1993shah committed:
Update ListDiskRestorePointInVmRestorePoint.json

reverting file changes

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json:
```diff
@@ -39,12 +46,18 @@
         "name": "myDisk",
         "location": "West US",
         "properties": {
-          "provisioningState": "Updating",
```
is promotionCode missing?

meet1993shah replied:
its optional, some purchase plan have it and some dont, I am not able to find the one where its present.

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json:
```diff
@@ -1,56 +1,48 @@
-    "subscriptionId": "{subscription-id}",
-    "resourceGroupName": "myResourceGroup",
+    "subscriptionId": "850f02f7-e14c-4fb7-905d-7b98bcae9c7f",
+    "resourceGroupName": "meeshahtest",
```
give a generic rg and subscription id..

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutASnapshot.json:
```diff
@@ -13,6 +13,12 @@
+            "name": "{VMPurchasePlanName}",
+            "publisher": "{VMPurchasePlanPublisherId}",
+            "product": "{VMPurchasePlanProduct}",
+            "PoromotionCode": "{VMPurchasePlanPromotionCode}"
```
typo: promotionCode

meet1993shah committed:
Update GetInformationAboutASnapshot.json

updated example for purchase plan

meet1993shah committed:
Update GetInformationAboutASnapshot.json

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

meet1993shah committed:
Update UpdateAManagedDiskToAddPurchasePlan.json

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

chetmaddula commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/GetInformationAboutAManagedDisk.json:
```diff
@@ -3,54 +3,46 @@
-        },
-        "type": "Microsoft.Compute/disks",
-        "location": "westus",
-        "tags": {
```
tags are missing

meet1993shah replied:
created the disk without tags, do I need to add tags?

chetmaddula replied:
yes, its supposed to show all the possible info in get disk only new properties need to be added, existing ones shouldnt be removed

meet1993shah replied:
oh I see okay, will update it.

meet1993shah committed:
Update GetInformationAboutAManagedDisk.json

meet1993shah committed:
Update GetInformationAboutASnapshot.json

laurawu19 commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/CreateAManagedDiskFromAPlatformImage.json:
```diff
@@ -11,7 +11,7 @@
           "createOption": "FromImage",
           "imageReference": {
-            "id": "/Subscriptions/{subscriptionId}/Providers/Microsoft.Compute/Locations/uswest/Publishers/Microsoft/ArtifactTypes/VMImage/Offers/{offer}"
+            "id": "/Subscriptions/{subscriptionId}/Providers/Microsoft.Compute/Locations/uswest/Publishers/{publisher}/ArtifactTypes/VMImage/Offers/{offer}/Skus/{sku}/Versions/1.0.0"
```
Locations/uswest? or Locations/westus? Respond should be the same as request.

meet1993shah committed:
Update CreateAManagedDiskFromAPlatformImage.json

vikramd-ms commented on specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/UpdateAManagedDiskToAddPurchasePlan.json:
```diff
@@ -0,0 +1,80 @@
+          "diskSizeGB": 127,
+          "provisioningState": "Succeeded",
+          "isArmResource": true,
+          "faultDomain": 0
```
isArmResource and faultDomain are internal only properties and should not appear in the response

meet1993shah committed:
Update UpdateAManagedDiskToAddPurchasePlan.json

removed isArmResource and faultDomain from the response
